src/python/gudhi/sklearn/cover_complex.py

```diff
@@ -0,0 +1,473 @@
+        self.graph, self.rips_threshold, self.N, self.beta, self.C = graph, rips_threshold, N, beta, C
+        self.distance_matrix_name, self.input_name, self.cover_name, self.color_name, self.verbose = distance_matrix_name, input_name, cover_name, color_name, verbose
+
+    def fit(self, X, y=None):
+        """
+        Fit the CoverComplex class on a point cloud or a distance matrix: compute the cover complex and store it in a simplex tree called simplex_tree.
+
+        Parameters:
+            X (numpy array of shape (num_points) x (num_coordinates) if point cloud and (num_points) x (num_points) if distance matrix): input point cloud or distance matrix.
+            y (n x 1 array): point labels (unused).
+        """
+        self.data = X
+
+        if self.colors is None:
+            if self.input_type == "point cloud":
+                self.colors = X[:,0] if self.complex_type == "gic" else X[:,0:1]
+            elif self.input_type == "distance matrix":
+                self.colors = X.max(axis=0) if self.complex_type == "gic" else X.max(axis=0)[:,np.newaxis]
+
+        if self.filters is None:
+            if self.input_type == "point cloud":
+                self.filters = X[:,0] if self.complex_type == "gic" else X[:,0:1]
+            elif self.input_type == "distance matrix":
+                self.filters = X.max(axis=0) if self.complex_type == "gic" else X.max(axis=0)[:,np.newaxis]
+
+        if self.complex_type == "gic" or self.complex_type == "nerve":
+
+            if self.complex_type is not "nerve" or self.cover is not "functional": # Nerve + functional cover is a special case where it's better to use Mapper than C++ code
+
+                ct = "GIC" if self.complex_type == "gic" else "Nerve"
+                self.complex = NGIComplex()
+                self.complex.set_type(ct)
+                self.complex.set_distance_matrix_name(self.distance_matrix_name)
+                self.complex.set_verbose(self.verbose)
+
+                if self.input_type == "point cloud":
+                    self.complex.set_point_cloud_from_range(X)
+                elif self.input_type == "distance matrix":
+                    self.complex.set_distances_from_range(X)
+            
+                self.complex.set_color_from_range(self.colors)
+            
+                if self.complex_type == "gic":
+                    if self.graph == "rips":
+                        if self.rips_threshold is not None:
+                            self.complex.set_graph_from_rips(self.rips_threshold)
+                        else:
+                            self.complex.set_subsampling(self.C, self.beta)
+                            self.complex.set_graph_from_automatic_rips(self.N)
+
+            if self.cover == "voronoi":
+                assert self.complex_type is not "nerve"
```

Comment:
`is not` should only be used for `None`, for instance. Use `!=` operator for literals.
```suggestion
                assert self.complex_type != "nerve"
```

Reply:
Thanks! Done.